test/Atmos/EDMF/helper_funcs/subdomain_thermo_states.jl

```diff
@@ -0,0 +1,235 @@
+#### thermo states for subdomains
+
+using KernelAbstractions: @print
+
+export new_thermo_state_up,
+    new_thermo_state_en,
+    recover_thermo_state_all,
+    recover_thermo_state_up,
+    recover_thermo_state_en
+
+####
+#### Interface
+####
+
+"""
+    new_thermo_state_up(bl, state, aux)
+
+Updraft thermodynamic states given:
+ - `bl`, parent `BalanceLaw`
+ - `state`, state variables
+ - `aux`, auxiliary variables
+
+!!! note
+    This method calls saturation adjustment for
+    EquilMoist models.
+
+# TODO:
+ Change recover_thermo_state to new_thermo_state
```

Comment:
This is done now?

Reply:
We should update the comment because this is perhaps more of a reminder about when/where we should call new thermo states-- a more general concern. I'm not sure if we should call `new_thermo_state` (for the grid mean) here, or if we rely on `recover_thermo_state` and ensure that the `aux` temperature has been updated before this call, which may be needed to fix upstream problems anyway.

Reply:
I think we need to handle this more systematically in a separate PR.